### PR TITLE
Home, End, Del operations for Input

### PR DESCRIPTION
**Pull request by alexzanderr**
Hello, I really like your crate, but is missing essential functionality.
Today I got the idea to implement only Home, End, Del operations for Input struct.

The only part that is important is:
```rust
                    Key::Home if position != 0 => {
                        term.move_cursor_left(position)?;
                        position = 0;
                        term.flush()?;
                    }
                    Key::End if position != chars.len() => {
                        term.move_cursor_right(chars.len() - position)?;
                        position = chars.len();
                        term.flush()?;
                    }
                    Key::Del
                        if 0 <= position && position < chars.len() =>
                    {
                        let tail = chars[position + 1..].iter().collect::<String>();

                        term.move_cursor_right(1)?;
                        term.clear_chars(1)?;
                        chars.remove(position);

                        term.write_str(&tail)?;
                        term.move_cursor_left(tail.len())?;

                        // if let Some(is_debug) = option_env!("DEBUG") {
                        //     term.write_line(&format!("\npos: {}", position))?;
                        //     term.write_line(&format!("\nlen: {}", chars.len()))?;
                        //     term.write_line(&format!("\nlen: {:?}", chars))?;

                        // }

                        term.flush()?;
                    },
```
The rest of modifications are formatting using `rustfmt`, is that is fine for you.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,4 @@
 target
 Cargo.lock
+
+examples/my_pr.rs
```
Comment (pksunkara) [inline .gitignore:4]: ?

File: examples/completion.rs [hunk 2]
```diff
@@ -21,6 +21,7 @@ impl Default for MyCompletion {
                 "orange".to_string(),
                 "apple".to_string(),
                 "banana".to_string(),
+                "banana2".to_string(),
```
Comment (pksunkara) [inline examples/completion.rs:24]: Why is this added?

File: examples/history.rs [hunk 3]
```diff
@@ -31,7 +31,7 @@ struct MyHistory {
 impl Default for MyHistory {
     fn default() -> Self {
         MyHistory {
-            max: 4,
+            max: 100,
```
Comment (pksunkara) [inline examples/history.rs:34]: ?

File: examples/input.rs [hunk 4]
```diff
@@ -1,4 +1,4 @@
-use dialoguer::{theme::ColorfulTheme, Input};
+    use dialoguer::{theme::ColorfulTheme, Input};
```
Comment (pksunkara) [inline examples/input.rs:1]: ?

**Review by pksunkara — CHANGES_REQUESTED** [review]
Please remove the unnecessary changes. Not sure why your rustfmt is behaving like that.

File: examples/completion.rs (hunk 2 above)
Comment (alexzanderr) [reply examples/completion.rs:24]: sorry I was playing around with the crate and forgot to delete the changes.

that is also available for other unnecessary changes

**Comment by alexzanderr** [comment]
sure.

about the rustfmt, i didnt see that this repo has a rustfmt standard. its missing a rustfmt.toml and basically there are no rules for formatting. 

so lets define one or just tell me you want the default formatting of rustfmt.

File: .gitignore (hunk 1 above)
Comment (alexzanderr) [reply .gitignore:4]: yeah. i will remove that

**Comment by pksunkara** [comment]
It's the default config.

**Comment by alexzanderr** [comment]
ok, the same as en empty rustfmt toml

**Comment by alexzanderr** [comment]
i will make another pr, this time a clean one, sorry for spam.
